core/domain/prod_validation_jobs_one_off.py

```diff
@@ -0,0 +1,151 @@
+            datastore_services.fetch_multiple_entities_by_ids_and_models(
+                multiple_models_keys_to_fetch.values()))
+
+        for (field_name, keys_to_fetch), model_class_models in zip(
+                multiple_models_keys_to_fetch.iteritems(),
+                fetched_model_instances):
+            model_class, model_ids = keys_to_fetch
+            for model_id, model in zip(model_ids, model_class_models):
+                if model is None or model.deleted:
+                    cls.errors['%s field check' % field_name] = (
+                        'Model id %s: based on field %s having'
+                        ' value %s, expect model %s with id %s but it doesn\'t'
+                        ' exist' % (
+                            item.id, field_name, model_id,
+                            str(model_class.__name__), model_id))
+
+
+class UserSubscriptionsModelValidator(BaseModelValidator):
+    """Class for validating UserSubscriptionsModels."""
+
+    @classmethod
+    def _external_id_relationships(cls):
+        return {
+            'activity_ids': exp_models.ExplorationModel,
+            'collection_ids': collection_models.CollectionModel,
+            'general_feedback_thread_ids': (
+                feedback_models.GeneralFeedbackThreadModel),
+            'creator_ids': user_models.UserSettingsModel,
+            'id': user_models.UserSettingsModel,
+        }
+
+    @classmethod
+    def validate(cls, item):
+        """Validates a UserSubscriptionsModel entity.
+
+        Args:
+            item: UserSubscriptionsModel. Entity to validate.
+        """
+        cls.errors.clear()
+
+        cls._validate_external_id_relationships(item)
+
+
+class ProdValidationAuditOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that audits and validates production models."""
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.UserSubscriptionsModel]
+
+    @staticmethod
+    def map(model_instance):
+        if not model_instance.deleted:
+            # Check if model_instance is a UserSubscriptionsModel here because
+            # we will add more model classes to iterate over in the future.
+            # TODO(sshou) remove this comment after we add other classes.
+            if isinstance(model_instance, user_models.UserSubscriptionsModel):
+                validate_user_subs_model = UserSubscriptionsModelValidator()
+                validate_user_subs_model.validate(model_instance)
+                if len(validate_user_subs_model.errors) > 0:
+                    for error_key, error_val in (
+                            validate_user_subs_model.errors.iteritems()):
+                        yield (
+                            'failed validation check for %s of '
+                            'UserSubscriptionModel' % error_key, error_val)
+                else:
+                    yield (
+                        'fully-validated UserSubscriptionModels', 1)
```

Comment:
Pull the string out into a class-level constant since you're using it in multiple places.

Reply:
I changed the reducer to yield succeed if 'fully-validated' in key.
In the next changes, I imagine we'll yield more general messages for e.g. model_instance.__cls__.__name__.